packages/flutter/lib/src/logging/logging.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/logging.dart';
+
+typedef void _RegisterServiceExtensionCallback(
+    { @required String name, @required ServiceExtensionCallback callback });
+
+/// Manages [Logger] services.
+class LoggingService {
+  /// The shared service instance.
+  static final LoggingService instance = new LoggingService._();
+
+  LoggingService._();
+
+  /// All the [Logger]s.
+  final Map<String, Logger> loggers = <String, Logger>{};
+
+  _RegisterServiceExtensionCallback _registerServiceExtensionCallback;
+
+  final Set<Logger> _subscriptions = new HashSet<Logger>();
+
+  /// Return `true` if the given [logger] has subscriptions.
+  bool hasSubscriptions(Logger logger) => _subscriptions.contains(logger);
+
+  /// Called to register service extensions.
+  void initServiceExtensions(
```

Comment:
service extension registrations should be in the binding

Reply:
My first cut had all of this in it's own binding class; would you prefer that?

FWIW: this is the angle taken by the `WidgetInspector`:

https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/widget_inspector.dart#L930

Reply:
Don't follow what WidgetInspector does. It's a very special case with weird requirements and constraints.